app/controllers/home_controller.rb

```diff
@@ -9,13 +9,17 @@ def index
 
   def check_card
     @card = Card.find(check_params[:card_id])
-    result = @card.check_translation(check_params[:user_translation])
-    case result[:typos_count]
-    when 0 then
+    result = @card.check_translation(@card,
+                                     check_params[:user_translation],
+                                     check_params[:response_time])
+    case result[:assession_mark]
+    when 5 then
       flash[:success] = t('card.check_success')
-    when 1, 2 then
+    when 4, 3 then
+      flash[:warning] = t('card.check_delay')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.